frontend/packages/console-app/src/components/quick-starts/utils/quick-start-context.ts

```diff
@@ -0,0 +1,116 @@
+import { QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY } from '../../../redux/reducers/quick-start-reducer';
+import { createContext, useCallback } from 'react';
+import { QuickStartStatus, QuickStartTaskStatus } from './quick-start-types';
+import { useUserSettings } from '@console/shared/src/hooks/useUserSettings';
+
+export type QuickStartContextValues = {
+  activeQuickStartID: string;
+  allQuickStartStates?: any;
+  activeQuickStartState?: any;
+  setActiveQuickStart?: (quickStartId: string, totalTasks?: number) => void;
+  setQuickStartStatus?: (quickStartId: string, status: QuickStartStatus) => void;
+  setQuickStartTaskNumber?: (quickStartId: string, taskNumber: number) => void;
+  setQuickStartTaskStatus?: (taskStatus: QuickStartTaskStatus) => void;
+  resetQuickStart?: (quickStartId: string, totalTasks: number) => void;
+  getQuickStartForId?: (id: string) => any;
+};
+
+export const QuickStartContext = createContext<QuickStartContextValues>({ activeQuickStartID: '' });
+
+const getInitialState = () =>
+  localStorage.getItem(QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY)
```

Comment:
Should we still use data present over localStorage or is it better to start with scratch as in the future this could lead to anomaly caused by a user with malicious intent.

Reply:
this is a temporary code once the local storage is migrated over to user settings this will be removed.